### ISSUE TITLE
Connectivity measure: going further

### ISSUE THREAD

**Issue by AlexandreAbraham** [nilearn/nilearn#1297]
At this moment, ConnectivityMeasure estimates a square connectivity matrix from timeseries. The next step, I think, would be to estimate connectivity directly from img. One way to do this is to have an object that takes one or two maskers and estimate the connectivity matrix between them (if one masker, estimate the connectivtiy of the extracted timeseries against themselves).

With this simple scheme, people will be able to easily do seed to ROI or ROI to voxel or ROI to ROI connectivity. Plus, if we use iterators, we can do this efficiently in memory. I think that could become a killer feature for nilearn and that shows how to leverage the maskers' power.

**Comment by GaelVaroquaux** [comment]
I'd rather have another object that does that. It would indeed combine
maskers, possiblyt dict learn and co, and connectivity measure

@KamalakerDadi has something like this internally for his paper.

**Comment by AlexandreAbraham** [comment]
> I'd rather have another object that does that.

Yes, I did not think of augmenting the ConnectivityMeasure class but adding a new one.

> @KamalakerDadi has something like this internally for his paper.

I know but what I had in mind is inbetween his work and what we have now. Plus, it has to be a specific object to be able to do it "online". Kamalakar could then re-use it in his work.

**Comment by GaelVaroquaux** [comment]
> I know but what I had in mind is inbetween his work and what we have now. Plus,
> it has to be a specific object to be able to do it "online". Kamalakar could
> then re-use it in his work.

In order not hasten design, I'd rather finish the experiments with the
object of Kamalakar, and draw experience from it first.

**Comment by KamalakerDadi** [comment]
From what I see. What might be missing currently be maskers memory consumption (particularly ), labels and maps maskers not accepting list of subjects and non-parallelization and confounds cleaning in Connectivity Measure. Apart from that, the object design idea looks really interesting.

>I know but what I had in mind is inbetween his work and what we have now. Plus, it has to be a specific object to be able to do it "online".

Can you elaborate on this please ? What is inbetween I am not able to strike and online ?
